### PR TITLE
CodeFuse IDE Support Web IDE #32

### PR DESCRIPTION
**Pull request by eedpublic**
#32  CodeFuse IDE Support Web IDE

### CODE REVIEW

File: patches/@opensumi+ide-core-browser+3.4.4.patch [hunk 1]
```diff
@@ -0,0 +1,109 @@
+diff --git a/node_modules/@opensumi/ide-core-browser/lib/common/common.contribution.js b/node_modules/@opensumi/ide-core-browser/lib/common/common.contribution.js
```
Comment (hacke2) [inline patches/@opensumi+ide-core-browser+3.4.4.patch:1]: 这个 patch 不要提交，直接往 opensumi 上提交 https://github.com/opensumi/core

Comment (eedpublic) [reply patches/@opensumi+ide-core-browser+3.4.4.patch:1]: 已经在这里提了 https://github.com/opensumi/core/pull/4160

File: README.md [hunk 2]
```diff
@@ -43,6 +43,16 @@ yarn run electron-rebuild
...
 ```
 
+### Start the project (web)
+```bash
+# install dependencies
+yarn
+# rebuild native dependencies for electron
+yarn run electron-rebuild
+# start project
+yarn run start-web
```
Comment (Cyrusky) [inline README.md:53]: There is no such script thing in `package.json`, is this a mistake?

File: src/web/browser/common-modules.ts
Comment (Cyrusky) [inline src/web/browser/common-modules.ts:1]: I think `bootstrap-web` is better than `web` as dir name.

File: src/web/browser/common-modules.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  ClientAddonModule,
+  CommentsModule,
+  TaskModule,
+  CoreBrowserModule,
+  TestingModule, RemoteOpenerModule,
+  // ai
+  DesignModule,
+  AINativeModule,
+  AIFeatureModule,
+  AutoUpdaterModule,
```
Comment (Cyrusky) [inline src/web/browser/common-modules.ts:94]: `AutoUpdateModule` is not necessary in a web environment.

File: src/web/browser/index.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import { ExpressFileServerModule } from '@opensumi/ide-express-file-server/lib/browser';
+import '@opensumi/ide-core-browser/lib/style/index.less';
+import '@opensumi/ide-core-browser/lib/style/icon.less';
+
+import { renderApp } from './render-app';
+import { CommonBrowserModules } from '../../../src/web/browser/common-modules';
+import { layoutConfig } from './layout-config';
+import './main.less';
+import './styles.less';
+import {AILayout} from "@opensumi/ide-ai-native/lib/browser/layout/ai-layout";
```
Comment (Cyrusky) [inline src/web/browser/index.ts:11]: Please follow `ESlint` and `Prettier` rules.

File: build/webpack-web/webpack.worker-host.config.ts
Comment (Cyrusky) [inline build/webpack-web/webpack.worker-host.config.ts:1]: By the way, Why mixed using the 'ts' and 'js' in the mean time ?

File: patches/@opensumi+ide-core-browser+3.4.4.patch (hunk 1 above)
Comment (hacke2) [reply patches/@opensumi+ide-core-browser+3.4.4.patch]: 那就把这块代码删掉

**Review by hacke2 — APPROVED** [review]
LGTM

**Pull request merged**